syntax = "proto3";
//import "google/protobuf/timestamp.proto";

package Core;

message TaskID {
  int32 value = 1;
}

message Task {
  string title = 1;
  enum Priority {
    NONE = 0;
    LOW = 1;
    MEDIUM = 2;
    HIGH = 3;
  }
  Priority priority = 2;
//  google.protobuf.Timestamp due_date = 3;
  int64  due_date = 3;
  repeated string labels = 4;
  bool is_complete = 5;
}

message TaskEntity {
  TaskID id = 1;
  Task data = 2;
  TaskID parent = 3;
}

message Label {
  string label = 1;
}

message ModelRequestResult {
    enum Status {
      SUCCESS = 0;
      ID_NOT_FOUND = 1;
      PARENT_ID_NOT_FOUND = 2;
      HAS_CHILDREN = 3;
      DUPLICATE_ID = 4;
      FILE_NOT_FOUND = 5;
      FAILED_TO_OPEN_FILE = 6;
      TAKES_NO_ARG = 7;
      TAKES_ARG = 8;
      TAKES_ID = 9;
      TAKES_ALPHA_NOT_ID = 10;
      TAKES_ALPHA = 11;
    };
    oneof data {
        Status status = 1;
        TaskID id = 2;
    }
}